### PR TITLE
Technical advisory 50587

### LINKED ISSUE #7673: Document a tech advisory 50587
https://docs.google.com/document/d/1LM63uTCTG1MPp0lnBZv5oQ5U8Ajd_andmGbXlxYf5wA/edit?usp=sharing

### PR DESCRIPTION
**Pull request by ericharmeling**
Fixes #7673.

### CODE REVIEW

File: _data/advisories.yml [hunk 1]
```diff
@@ -1,3 +1,7 @@
+- advisory: 50587
```
Comment (dbist) [inline _data/advisories.yml:1]: advisory 50587 and 50766

File: _data/advisories.yml [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
+  summary: TRUNCATE prevents table renaming
+  versions: 2.1.0-2.1.9, 19.1.0-19.1.10, 19.2.0 - 19.2.8
```
Comment (dbist) [inline _data/advisories.yml:3]: I don't think we care about 2.1.x anymore, I'd safely remove, especially that a backport to 2.1 is not happening.

File: _data/advisories.yml [hunk 3]
```diff
@@ -1,3 +1,7 @@
...
+  date: June 29, 2020
```
Comment (dbist) [inline _data/advisories.yml:4]: I would change the date to the date of the maintenance release

File: advisories/a50587.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+---
+title: Technical Advisory 50587
```
Comment (dbist) [inline advisories/a50587.md:2]: Technical Advisory 50587, 50766

File: advisories/a50587.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+summary: TRUNCATE prevents table renaming
+toc: true
+---
+
+Publication date: June 29, 2020.
```
Comment (dbist) [inline advisories/a50587.md:7]: synch the publication date to the release date

File: advisories/a50587.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+## Description
+
+When executing a [`TRUNCATE`](https://www.cockroachlabs.com/docs/stable/truncate.html) statement on table, CockroachDB creates a new table, with a new descriptor ID, copies the contents from the old table to the new table, and then [drops](https://www.cockroachlabs.com/docs/stable/drop-table.html) the old table. When the old table is dropped, a [drop job](https://www.cockroachlabs.com/docs/stable/online-schema-changes.html) is created to asynchronously clean up the table's data. In v19.2, v19.1, and prior versions, a job ID is assigned to the old table prior to copying its contents to the new table. This means that the new table descriptor will have a populated drop job ID. This drop job ID prevents the table from being [renamed](https://www.cockroachlabs.com/docs/v20.1/rename-table.html).
+
+## Statement
+
+This bug has been fixed in versions v19.1.11, v19.2.9, and v20.1.0. All users are encouraged to upgrade (see [Mitigation](#mitigation) below).
```
Comment (dbist) [inline advisories/a50587.md:15]: I am not sure the release versions of the said maintenance releases. v20.1 is not impacted by this bug.

File: advisories/a50587.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+The public issue is tracked as [#50587](https://github.com/cockroachdb/cockroach/issues/50587).
```
Comment (dbist) [inline advisories/a50587.md:17]: please check the latest doc, it has link to the PR for 50766 which targets 19.1, 50587 is for 19.2 fix only.

**Comment by ericharmeling** [comment]
Release notes:

https://github.com/cockroachdb/docs/pull/7690

**Pull request merged**